Source/CarthageKit/BuildArguments.swift

```diff
@@ -84,6 +84,11 @@ public struct BuildArguments {
 			args += [ "BITCODE_GENERATION_MODE=\(bitcodeGenerationMode.rawValue)" ]
 		}
 
+		if let carthagePath = NSBundle.mainBundle().executablePath
+			where NSBundle.mainBundle().executableURL?.lastPathComponent?.lowercaseString.rangeOfString("carthage") != nil {
```

Comment:
Can you even do that? The `carthage` binary relies on CarthageKit.framework, which is installed in /Library/Frameworks. I don't see how you can possibly have multiple versions of Carthage installed unless you put them in different directories with a copy of CarthageKit.framework in the same directory (that's the only way to get it to load a different version of CarthageKit.framework than the one in /Library/Frameworks). And if they're in different directories then there's no reason to rename the executable.